### DISCUSSION TITLE
Drag Iframe Element in CDP Mode

### DISCUSSION THREAD

**Discussion by harrier999** [seleniumbase/SeleniumBase#3302] [Q&A] [answered]
Is it possible to drag iframe element in CDP mode?
`get_nested_element` retrives element in iframe. But it seems `mouse_drag` method does not work with the element

Or is there relative drag method for element in CDP mode?

**Comment by mdmintz** [comment, marked as answer]
To drag and drop an element in [CDP Mode](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/cdp_mode/ReadMe.md):

```python
sb.cdp.gui_drag_and_drop(drag_selector, drop_selector)
```

Example in [examples/cdp_mode/raw_demo_site.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/cdp_mode/raw_demo_site.py)

**Reply by harrier999** [reply]
1. There are no examples of drag and drop inside an iframe there.

2. This does not work
```python
handle = sb.cdp.get_nested_element(iframe, hadle_selector)
position = handle.get_position()
x1, y1 = position.x, position.y
x2, y2 = x1 + 50, y1,
sb.cdp.gui_drag_drop_points(x1, y1, x2, y2)
```

3. What if the element doesn't exist in the destination?
How do I move the starting point by x and y pixels relative to the destination?

**Reply by mdmintz** [reply]
You'll have to use `sb.cdp.gui_drag_drop_points(x1, y1, x2, y2)` and determine the coordinates yourself.
